packages/docs-app/src/examples/select-examples/multiSelectExample.tsx

```diff
@@ -172,7 +166,7 @@ export class MultiSelectExample extends BaseExample<IMultiSelectExampleState> {
 
     private handleSwitchChange(prop: keyof IMultiSelectExampleState) {
         return (event: React.FormEvent<HTMLInputElement>) => {
-            this.setState({ [prop]: event.currentTarget.checked });
+            this.setState({ ...this.state, [prop]: event.currentTarget.checked });
```

Comment:
This is bad. Either use `this.setState(state => { …state… })` or, in this case, don't spread state at all.

Reply:
i did this cuz the typings didn't like the `[prop]` key, despite the fact that it's defined as one of the valid keys. thoughts?

Reply:
Not every key in the state is a `boolean`, so that error makes sense. Have you tried creating a type that only includes the boolean keys?

Reply:
@invliD i actually cannot get this to typecheck. i've tried all sorts of Picks, Partials, Records, and it always comes back to this error. the spread is the only workaround i've found thus far.

Reply:
oh hey `as any` works great 🌴